winpty/src/windows.rs

```diff
@@ -55,16 +57,30 @@ fn check_err(e: *mut winpty_error_t) -> Option<Err> {
         let raw = winpty_error_msg(e);
         let message = String::from_utf16_lossy(std::slice::from_raw_parts(raw, wcslen(raw)));
         winpty_error_free(e);
+
+        let code = match code {
+            WINPTY_ERROR_SUCCESS => ErrorCode::Success,
+            WINPTY_ERROR_OUT_OF_MEMORY => ErrorCode::OutOfMemory,
+            WINPTY_ERROR_SPAWN_CREATE_PROCESS_FAILED => ErrorCode::SpawnCreateProcessFailed,
+            WINPTY_ERROR_LOST_CONNECTION => ErrorCode::LostConnection,
+            WINPTY_ERROR_AGENT_EXE_MISSING => ErrorCode::AgentExeMissing,
+            WINPTY_ERROR_UNSPECIFIED => ErrorCode::Unspecified,
+            WINPTY_ERROR_AGENT_DIED => ErrorCode::AgentDied,
+            WINPTY_ERROR_AGENT_TIMEOUT => ErrorCode::AgentTimeout,
+            WINPTY_ERROR_AGENT_CREATION_FAILED => ErrorCode::AgentCreationFailed,
+            code => ErrorCode::UnknownError(code)
+        };
+
         match code {
-            0 => None,
-            _ => Some(Err { code, message }),
+            ErrorCode::Success => None,
+            code => Some(Err { code, message }),
```

Comment:
I love matching against `ErrorCode::Success` here, instead of `0`. Great change!

Reply:
👍 - though this code will now die to eliminate the double-match 😄